test/impl_test.py

```diff
@@ -92,198 +138,271 @@ def getType(self, ref=None):
-                        "ERROR: Invalid Genome reference {} from {}".format(ret[0], self.genome_ref))
+        self.assertTrue(
+            self.getType(ret[0]).startswith("KBaseGenomeAnnotations.Taxon"),
+            "ERROR: Invalid Genome reference {} from {}".format(ret[0],
+                                                                self.old_genome_ref))
 
     @log
     def test_get_assembly(self):
-        inputs = {'ref': self.genome_ref}
+        inputs = {'ref': self.new_genome_ref}
         ret = self.impl.get_assembly(self.ctx, inputs)
         self.assertTrue(self.getType(ret[0]).startswith("KBaseGenomeAnnotations.Assembly"),
-                        "ERROR: Invalid ContigSet reference {} from {}".format(ret[0], self.genome_ref))
+                        "ERROR: Invalid assembly reference {} from {}".format(ret[0], self.new_genome_ref))
+
+    @log
+    def test_get_assembly_old(self):
+        inputs = {'ref': self.new_genome_ref}
+        ret = self.impl.get_assembly(self.ctx, inputs)
+        self.assertTrue(
+            self.getType(ret[0]).startswith("KBaseGenomeAnnotations.Assembly"),
+            "ERROR: Invalid assembly reference {} from {}".format(ret[0],
+                                                                   self.old_genome_ref))
 
     @log
     def test_get_feature_types(self):
-        inputs = {'ref': self.genome_ref}
-        ret = self.impl.get_feature_types(self.ctx, inputs)
-        self.assertGreater(len(ret[0]), 0, "ERROR: No feature types present {}".format(self.genome_ref))
+        with self.assertRaises(NotImplementedError):
+            inputs = {'ref': self.new_genome_ref}
+            ret = self.impl.get_feature_types(self.ctx, inputs)
 
     @log
     def test_get_feature_type_descriptions_all_old(self):
-        inputs = {'ref': self.genome_ref}
-        ret = self.impl.get_feature_type_descriptions(self.ctx, inputs)
-        self.assertGreater(len(ret[0].keys()), 0, "ERROR: Feature type descriptions empty {}".format(self.genome_ref))
+        with self.assertRaises(NotImplementedError):
+            inputs = {'ref': self.new_genome_ref}
+            ret = self.impl.get_feature_type_descriptions(self.ctx, inputs)
 
     @log
     def test_get_feature_type_counts_all(self):
-        inputs = {'ref': self.genome_ref}
-        ret = self.impl.get_feature_type_counts(self.ctx, inputs)
-        self.assertGreater(len(ret[0].keys()), 0, "ERROR: Feature type counts empty {}".format(self.genome_ref))
+        with self.assertRaises(NotImplementedError):
+            inputs = {'ref': self.new_genome_ref}
+            ret = self.impl.get_feature_type_counts(self.ctx, inputs)
 
     @log
     def test_get_feature_ids_all(self):
-        inputs = {'ref': self.genome_ref}
+        inputs = {'ref': self.new_genome_ref}
+        ret = self.impl.get_feature_ids(self.ctx, inputs)
+        self.assertGreater(len(ret[0]), 0, "ERROR: No feature ids returned for all {}".format(self.new_genome_ref))
```

Comment:
This doesn't necessary check your return content is expected. Is there a reason why you only want to check the length instead of list content? Also, here do you mean ret['by_type'] instead of ret[0]?